src/Type/Symfony/ParameterDynamicReturnTypeExtension.php

```diff
@@ -187,18 +194,31 @@ private function getHasTypeFromMethodCall(
 		Scope $scope
 	): Type
 	{
-		$returnType = ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+		$defaultReturnType = ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
 		if (!isset($methodCall->getArgs()[0]) || !$this->constantHassers) {
-			return $returnType;
+			return $defaultReturnType;
 		}
 
-		$parameterKey = $this->parameterMap::getParameterKeyFromNode($methodCall->getArgs()[0]->value, $scope);
-		if ($parameterKey !== null) {
+		$parameterKeys = $this->parameterMap::getParameterKeysFromNode($methodCall->getArgs()[0]->value, $scope);
+		if ($parameterKeys === []) {
+			return $defaultReturnType;
+		}
+
+		$has = null;
+		foreach ($parameterKeys as $parameterKey) {
 			$parameter = $this->parameterMap->getParameter($parameterKey);
-			return new ConstantBooleanType($parameter !== null);
+
+			if ($has === null) {
+				$has = $parameter !== null;
+			} elseif (
+				$has === true && $parameter === null
+				|| $has === false && $parameter !== null
```

Comment:
Parentheses here please to disambiguate the logic.

Reply:
Sure. Done.